packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -509,7 +509,7 @@ class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay
               child: widget.selectionControls.buildHandle(
                 context,
                 type,
-                widget.renderObject.size.height / widget.renderObject.maxLines,
+                widget.renderObject.preferredLineHeight,
```

Comment:
Let's use this in line 347 too. (The material toolbar is a little buggy too on multi-line text fields)

Reply:
done